src/systems/sprites.js

```diff
@@ -187,224 +177,234 @@ function createGeometry(maxSprites) {
+      for (const sprite of sprites) {
+        const o = sprite.el.object3D;
         if (isVisible(o)) {
           raycastOnSprite.call(o, raycaster, intersects);
         }
       }
     }
-
     return intersects;
   }
-  constructor(scene) {
-    this.missingSprites = [];
-    this.maxSprites = 1024;
-    this.slots = { action: new Array(this.maxSprites), notice: new Array(this.maxSprites) };
-    this.spriteWithIndex = { action: new Map(), notice: new Map() };
-    this.indexWithSprite = { action: new Map(), notice: new Map() };
-    this.stack = { action: new Array(this.maxSprites), notice: new Array(this.maxSprites) };
-    this.meshes = {};
-
-    for (const stack of Object.values(this.stack)) {
-      for (let i = 0; i < this.maxSprites; i++) {
-        stack[i] = this.maxSprites - 1 - i;
-      }
-    }
 
-    const vertexShader = String.prototype.concat(
-      scene.renderer.xr.multiview ? multiviewVertPrefix : nonmultiviewVertPrefix,
-      vert
-    );
-    const fragmentShader = String.prototype.concat(
-      scene.renderer.xr.multiview ? multiviewFragPrefix : nonmultiviewFragPrefix,
-      frag
-    );
-
-    const domReady = waitForDOMContentLoaded();
-
-    Promise.all([domReady]).then(() => {
-      for (const [spritesheetPng, type] of pngs) {
-        Promise.all([createImageTexture(spritesheetPng, getThemeColorShifter(type)), waitForDOMContentLoaded()]).then(
-          ([spritesheetTexture]) => {
-            const material = new THREE.RawShaderMaterial({
-              uniforms: {
-                u_spritesheet: { value: spritesheetTexture },
-                hubs_Time: { value: 0 }
-              },
-              vertexShader,
-              fragmentShader,
-              side: THREE.DoubleSide,
-              transparent: true
-            });
-            const mesh = (this.meshes[type] = new THREE.Mesh(createGeometry(this.maxSprites), material));
-            const el = document.createElement("a-entity");
-            el.classList.add("ui");
-            scene.appendChild(el);
-            el.setObject3D("mesh", mesh);
-            mesh.frustumCulled = false;
-            mesh.renderOrder = window.APP.RENDER_ORDER.HUD_ICONS;
-            mesh.raycast = this.raycast.bind(this);
-          }
-        );
+  constructor(scene) {
+    this.sprites = { action: [], notice: [] };
+    this.indicesFromSprites = { action: new WeakMap(), notice: new WeakMap() };
+    this.meshes = { action: null, notice: null };
+
+    waitForDOMContentLoaded().then(() => {
+      for (const type in PNGS) {
+        const spritesheetPng = PNGS[type];
+        createImageTexture(spritesheetPng, getThemeColorShifter(type)).then(spritesheetTexture => {
+          const geometry = createGeometry(MAX_SPRITES);
+          const material = createMaterial(spritesheetTexture);
+          const mesh = (this.meshes[type] = new THREE.Mesh(geometry, material));
+          const el = document.createElement("a-entity");
```

Comment:
Minor since its just 2 entities total, but I wonder if we actually need this? We should be able to just add the Mesh to the scene directly.

Reply:
Does it mean

```
const mesh = (this.meshes[type] = new THREE.Mesh(geometry, material));
scene.object3D.add(mesh);
...
```

?

I just copied the block from the current code and I'm not sure if it's ok because I'm not familiar with A-Frame design.